backend/src/main/java/kr/momo/service/meeting/MeetingConfirmService.java

```diff
@@ -0,0 +1,94 @@
+
+@Service
+@RequiredArgsConstructor
+public class MeetingConfirmService {
+
+    private final MeetingRepository meetingRepository;
+    private final AttendeeRepository attendeeRepository;
+    private final AvailableDateRepository availableDateRepository;
+    private final ConfirmedMeetingRepository confirmedMeetingRepository;
+
+    @Transactional
+    public void create(String uuid, long attendeeId, MeetingConfirmRequest request) {
+        LocalDateTime startDateTime = request.startDateTime();
+        LocalDateTime endDateTime = request.endDateTime();
+
+        Meeting meeting = meetingRepository.findByUuid(uuid)
+                .orElseThrow(() -> new MomoException(MeetingErrorCode.INVALID_UUID));
+
+        Attendee attendee = attendeeRepository.findByIdAndMeeting(attendeeId, meeting)
+                .orElseThrow(() -> new MomoException(AttendeeErrorCode.INVALID_ATTENDEE));
+        validateHostPermission(attendee);
+
+        if (confirmedMeetingRepository.existsByMeeting(meeting)) {
+            throw new MomoException(ConfirmedScheduleErrorCode.ALREADY_EXIST_CONFIRMED_SCHEDULE);
+        }
+
+        validateMeetingLocked(meeting);
+        validateTimeRange(meeting, startDateTime, endDateTime);
+        validateDateRange(meeting, startDateTime, endDateTime);
+
+        ConfirmedMeeting confirmedMeeting = new ConfirmedMeeting(meeting, startDateTime, endDateTime);
+        confirmedMeetingRepository.save(confirmedMeeting);
+    }
+
+    private void validateHostPermission(Attendee attendee) {
+        if (attendee.isNotHost()) {
+            throw new MomoException(AttendeeErrorCode.ACCESS_DENIED);
+        }
+    }
+
+    private void validateMeetingLocked(Meeting meeting) {
+        if (!meeting.isLocked()) {
+            throw new MomoException(MeetingErrorCode.MEETING_UNLOCKED);
+        }
+    }
+
+    private void validateTimeRange(Meeting meeting, LocalDateTime startDateTime, LocalDateTime endDateTime) {
+        if (startDateTime.isAfter(endDateTime)
+                || !meeting.isContainedWithinTimeRange(startDateTime.toLocalTime(), endDateTime.toLocalTime())) {
+            throw new MomoException(ConfirmedScheduleErrorCode.INVALID_DATETIME_RANGE);
+        }
+    }
+
+    private void validateDateRange(Meeting meeting, LocalDateTime startDateTime, LocalDateTime endDateTime) {
+        AvailableDates availableDates = new AvailableDates(availableDateRepository.findAllByMeeting(meeting));
+        LocalDate startDate = startDateTime.toLocalDate();
+        LocalDate endDate = endDateTime.toLocalDate();
+
+        if (startDate.equals(endDate)) {
+            if (availableDates.notExistsByDate(startDate)) {
+                throw new MomoException(ConfirmedScheduleErrorCode.INVALID_DATETIME_RANGE);
+            }
+            return;
+        }
+
+        if (meeting.isNotFullTime()) {
+            throw new MomoException(ConfirmedScheduleErrorCode.INVALID_DATETIME_RANGE);
+        }
```

Comment:
시작일이 동일한 경우와 동일하지 않은 경우로 나눠서 로직을 구현하고 싶으셨던 것 같아요!
Early Return 으로 depth를 늘리기 보단 풀어서 설명해주시는게 가독성이 좋을 것 같아요!
```suggestion
        if (startDate.equals(endDate) && availableDates.notExistsByDate(startDate)) {
            throw new MomoException(ConfirmedScheduleErrorCode.INVALID_DATETIME_RANGE);
        }

        if (!startDate.equals(endDate) && meeting.isNotFullTime()) {
            throw new MomoException(ConfirmedScheduleErrorCode.INVALID_DATETIME_RANGE);
        }

        if (availableDates.isContainedWithinDateRange(startDate, endDate)) {
            throw new MomoException(ConfirmedScheduleErrorCode.INVALID_DATETIME_RANGE);
        }
```

Reply:
좋네욥👍 동의합니다.